app/data_client.go

```diff
@@ -109,6 +109,37 @@ type TabularData struct {
 	TimeReceived  time.Time
 }
 
+// ExportTabularDataReturn represents the result of an ExportTabularData API call.
+type ExportTabularDataReturn struct {
+	OrganizationID   string
+	LocationID       string
+	RobotID          string
+	RobotName        string
+	PartID           string
+	PartName         string
+	ComponentName    string
+	ComponentType    string
+	MethodName       string
+	TimeCaptured     time.Time
+	MethodParameters map[string]*GenericProtoMessage
+	Tags             []string
+	Payload          map[string]interface{}
+}
+
+// ExportTabularDataStream is a stream that returns ExportTabularDataReturns.
+type ExportTabularDataStream struct {
+	Stream pb.DataService_ExportTabularDataClient
+}
+
+// Next gets the next ExportTabularDataReturn.
+func (e *ExportTabularDataStream) Next() (*ExportTabularDataReturn, error) {
```

Comment:
I can't claim to be an expert on writing Go SDKs, so this question might be more for my own understanding. How did we arrive at creating a `Next` method? Is this preferable over something like returning a channel?

Reply:
I believe the idea is for the user to keep calling next to get readings as a stream. I based this on what we do for the robot logs streaming API and how it's implemented in the Go SDK.

Reply:
We're giving the user the option of what to do with the stream. Based on my conversation with others who are more familiar with Go standards, it's not common for Go to force users into a certain method of using something. The user can turn it into their own channel if that's what they prefer.